Automattic/happychat-client
# Stop chat pane from getting overly tall in Chrome

Addressing 14589-gh-Automattic/woocommerce.com – HappyChat client on WooCommerce.com is getting inordinately big in some browsers. 

I'm not able to reproduce this exactly, and I still have no idea what causes it – although I've observed two possibly related phenomena:

- If the `I need help with:` search input dropdown list is very tall, it makes the iframe a lot taller. 
- If you remove the max height on `.happychat__page` some JS keeps incrementing the iframe's `height` attribute.

## Changes in this PR

- Changed max height of `.happychat__page` from `99vh` to `768px`, to avoid chat pane getting too tall.
- Removed `18vh` height of `.happychat__composer`, to avoid composer pane getting too tall.
- Changed flex grow rule for `.happychat__conversation::after` element, so messages align to the bottom of the conversation window.
- Replaced `node-sass` with `sass` – M1 macs don't support `node-sass` very well.
- Updated version to 0.1.7.
- Related [diff](D89154-code) and [WCCOM PR](14616-gh-Automattic/woocommerce.com).

## Testing

- Log in to [Happy Chat staging environment](https://hud-staging.happychat.io). Accept chats and make sure you have English, WordPress.com and WooCommerce.com selected in Preferences.
- If you don't have  a file `src/config/development.local.json`, create it, and paste this in:

```json
{
  "css_url": "http://localhost:9000/"
}
```

- Run `npm install && npm start`.
- View http://localhost:9000/example-woo.html in Chrome.
- Open dev tools and add a rule `height: 1500px; outline: 1px solid red;` to the iframe, to mimic an overly tall container.
- Enter some messages in the chat window. The conversation should be aligned to the bottom. The chat pane should be no taller than 768px.
- If you want to see the initial support request form in the Happy Chat client again, end the chat in Happy Chat with the command `/end`.
- If for some reason you can't get the chat session to start, add this around line 30 of `targets/standalone/example-woo.html`:

```
canChat: true,
entry: 'ENTRY_CHAT',
```

## Deployment

D89154-code

## Screenshots

#### Possible cause of tall container

<img width="500" alt="image" src="https://user-images.githubusercontent.com/1647564/194265133-d0bb792b-edc0-4ad9-9899-9f2da3c4d857.png">

#### Chat pane is limited to 768px high, composer pane is about 66px high, and messages are aligned at the bottom

<img width="500" alt="image" src="https://user-images.githubusercontent.com/1647564/194860034-2aa0ff03-c97a-48d2-9156-f6ce838eb200.png">
